frontend-react/src/utils/misc.test.ts

```diff
@@ -218,3 +219,25 @@ describe("isProhibitedName", () => {
         });
     });
 });
+
+describe("parseFileLocation", () => {
+    test("returns the folder location, sending org, and filename when valid url", () => {
+        const { folderLocation, sendingOrg, fileName } = parseFileLocation(
+            "https://azurite:10000/devstoreaccount1/reports/receive%2Fsimple_report.csvuploader%2Fupload-covid-19-c33f9d36-9e5b-44eb-9368-218d88f3a7d1-20230131190253.csv"
+        );
+        expect(folderLocation).toEqual("RECEIVE");
+        expect(sendingOrg).toEqual("simple_report.csvuploader");
+        expect(fileName).toEqual(
+            "upload-covid-19-c33f9d36-9e5b-44eb-9368-218d88f3a7d1-20230131190253.csv"
+        );
+    });
+
+    test("returns empty strings when string is missing three '/'", () => {
```

Comment:
Maybe another test that'll verify that all three strings are empty when there aren't three fragments split on `%2F`?